servicetalk-loadbalancer-experimental/src/main/java/io/servicetalk/loadbalancer/NoopOutlierDetector.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright © 2024 Apple Inc. and the ServiceTalk project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.servicetalk.loadbalancer;
+
+import io.servicetalk.client.api.LoadBalancedConnection;
+import io.servicetalk.concurrent.api.Executor;
+
+import java.util.concurrent.TimeUnit;
+
+import static java.util.Objects.requireNonNull;
+
+// A simple outlier detector implementation that only provides the basic `RequestTracker` implementation
+// so the P2C LoadBalancingPolicy can still be effective.
+final class NoopOutlierDetector<ResolvedAddress, C extends LoadBalancedConnection>
```

Comment:
This could be a premature optimization: We can probably do without this and just use the XdsOutlierDetector with everything disable if we wanted. This does make the testing easier: otherwise a few tests pop because they expect a certain number of timer tasks and the xds detector makes one.

Reply:
Is it really NoOp then?